runtime/codert_vm/cnathelp.cpp

```diff
@@ -1429,31 +1429,24 @@ old_fast_jitResolvedFieldIsVolatile(J9VMThread *currentThread)
 static VMINLINE void*
 old_slow_jitResolveFieldImpl(J9VMThread *currentThread, UDATA parmCount, J9ConstantPool *ramConstantPool, UDATA cpIndex, void *jitEIP, bool isSetter, bool direct)
 {
-	void *addr = NULL;
-	J9RAMFieldRef *ramFieldRef = (J9RAMFieldRef*)ramConstantPool + cpIndex;
-	UDATA valueOffset = ramFieldRef->valueOffset;
-	UDATA flags = ramFieldRef->flags;
-	if (J9_UNEXPECTED(!VM_VMHelpers::instanceFieldRefIsResolved(flags, valueOffset, isSetter))) {
```

Comment:
Why don't we check if the CP entry is already resolved?

Reply:
Rather than duplicate the complexity of the final check, it's much simpler to just run the resolve.